sudoku-django-project/mainapp/classes.py

```diff
@@ -0,0 +1,284 @@
+from random import shuffle, randint
+import typing
+
+BoardType = typing.List[typing.List[typing.Union[None, int]]]    # type hint alias for board - a list of lists
+                                                                 # with int or None as values
+class Sudoku:
+    """" A class that acts like a sudoku puzzle. """
+
+    def __init__(self):  # noqa: ANN204
+        self.counter = 1
+        self.top_boxes = [_Box() for _ in range(3)]
+        self.mid_boxes = [_Box() for _ in range(3)]
+        self.bottom_boxes = [_Box() for _ in range(3)]
+
+        self._set_box_locations()
+
+        self.original = [[1, 2, 3, 4, 5, 6, 7, 8, 9],
+                         [4, 5, 6, 7, 8, 9, 1, 2, 3],
+                         [7, 8, 9, 1, 2, 3, 4, 5, 6],
+                         [2, 3, 1, 5, 6, 4, 8, 9, 7],
+                         [5, 6, 4, 8, 9, 7, 2, 3, 1],
+                         [8, 9, 7, 2, 3, 1, 5, 6, 4],
+                         [3, 1, 2, 6, 4, 5, 9, 7, 8],
+                         [6, 4, 5, 9, 7, 8, 3, 1, 2],
+                         [9, 7, 8, 3, 1, 2, 6, 4, 5]]
+
+        self.generator()
+        self.full_board = self.get_all_row_values()
+        self.puzzle_maker()
+
+    # Methods run at initialization.
+    def _set_box_locations(self) -> None:  # Sets each of the box's position attribute.
```

Comment:
Can remove, I don't think it breaks anything and is unnecessary

Reply:
Also put all the methods run at initialization here, I think that makes it more organized

Reply:
alright 👌